### PR TITLE
no throws error after close debugger

### LINKED ISSUE #227: Closing the debugger when there is no active notebook throws an error
Closing the debugger when there is no active notebook throws an error.

This seems to be because we call the `debug.stop()` method when the model is disposed:

https://github.com/jupyterlab/debugger/blob/57951d6cfb065f16c631a281fca1dfb77f0d0b09/src/index.ts#L112

Do we need to stop the debugger?

To reproduce:

1. Open the debugger
2. Open a notebook
3. Close the notebook
4. Close the debugger

debugger-close-no-notebook

### PR DESCRIPTION
**Pull request by KsavinN**
Resolve #227

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -109,11 +109,14 @@ class DebuggerHandler<H extends ConsoleHandler | NotebookHandler> {
     });
 
     debug.model.disposed.connect(async () => {
-      await debug.stop();
-      Object.keys(this.handlers).forEach(id => {
-        this.handlers[id].dispose();
-      });
-      this.handlers = {};
+      const idsOfHandler = Object.keys(this.handlers);
```
Comment (jtpio) [inline src/index.ts]: How about `handlerIds` instead of `idsOfHandler`?

File: src/index.ts [hunk 2]
```diff
@@ -109,11 +109,14 @@ class DebuggerHandler<H extends ConsoleHandler | NotebookHandler> {
...
+      if (idsOfHandler.length > 0) {
+        await debug.stop();
+        idsOfHandler.forEach(id => {
+          this.handlers[id].dispose();
+        });
+        this.handlers = {};
+      }
```
Comment (jtpio) [inline src/index.ts]: How about using an early return if there is no handler?

```typescript
if (handlerIds.length === 0) {
  return;
}
await debug.stop();
handlerIds.forEach(id => {
  this.handlers[id].dispose();
});
this.handlers = {};

```

File: src/index.ts (hunk 1 above)
Comment (KsavinN) [reply src/index.ts]: Resolved

File: src/index.ts (hunk 2 above)
Comment (KsavinN) [reply src/index.ts]: Resolved

**Pull request merged**